File: Project Οντοκεντρικός Προγραμματισμός/C++ Project/CodeBlocks Project Hotel/src/RoomTypeC.cpp
#include <RoomTypeC.h>

#include<Booking.h>

int minPeople;//Variable used for the minimum people of a room type C.
int minDays;//Variable used for the minimum days of a room type C.

RoomTypeC::RoomTypeC()//Default constructor for a RoomTypeC type object.
{
}

RoomTypeC::RoomTypeC(int minPeople, int minDays,int maxPeople,double pricePerPerson)//Constructor for a RoomTypeC type object.
{
	setMinPeople(minPeople);
	setMinDays(minDays);
	setMaxPeople(maxPeople);
	setPricePerPerson(pricePerPerson);
}

RoomTypeC::~RoomTypeC()//Default destructor for a RoomTypeC type object.
{
}

bool RoomTypeC::addBooking(Booking *a)//Function that returns true if you can book in a room type C or false if not.
{
	int i;
	for (i = a->getArrivalDay(); i <= a->getArrivalDay() + a->getAccomodationDays() - 1; i++)
	{
		if (roomAvailability[i] != 0 || a->getPeopleNum() > this->getMaxPeople() || a->getPeopleNum() < this->minPeople || a->getAccomodationDays() < this->minDays)
			return false;
	}
	for (i = a->getArrivalDay(); i <= a->getArrivalDay() + a->getAccomodationDays() - 1; i++)
	{
		roomAvailability[i] = a;
	}
	a->setRoom(this);//Change the Room type pointer form 0(null pointer) to this RoomTypeC.
	return true;
}

int getMinPeople()//Getter for the variable minPeople.
{
	return minPeople;
}

void setMinPeople(int val)//Setter for the variable minPeople.
{
	minPeople = val;
}

int getMinDays()//Getter for the variable minDays.
{
	return minDays;
}

void setMinDays(int val)//Setter for the variable minDays.
{
	minDays = val;
}

File: Project Οντοκεντρικός Προγραμματισμός/Java Project/Hotel Project, Swing/RoomTypeC.java
public class RoomTypeC extends Room
{
    private static int minPeople=5; //Μεταβλητή για ελάχιστα άτομα στο δωμάτιο τύπου C.
    private static int minDays=5;  //Μεταβλητή για ελάχιστες μέρες στο δωμάτιο τύπου C.
    //Getters
    public static int getMinDays()
    {
        return minDays;
    }
     public static int getMinPeople()
    {
        return minPeople;
    }
    //Setters
    public static void setMinDays(int ab)
    {
        minDays = ab;
    }
    public static void setMinPeople(int abbb)
    {
        minPeople = abbb;
    }
    public boolean  addBooking (Booking a)//Ειδική μέθοδος για την προσθήκη κράτησης,για δωμάτιο τύπου C,με βάση τον
    {                                     //αριθμό τον ελάχιστων ατόμων και ημερών διαμονής.
        int i=0;
        for(i=a.getArrivalDate(); i < a.getArrivalDate()+a.getAccomodationDays()-1; i++)
        {
            if(roomAvailability[i]!=null || a.getPeopleNum()>getMaxPeople() || a.getPeopleNum() < minPeople || a.getAccomodationDays() < minDays)
            {
                return false;   
            }
        }
        for(i=a.getArrivalDate(); i <= a.getArrivalDate()+a.getAccomodationDays() - 1; i++)
        {
            roomAvailability[i] = a;
        }
        a.setRoom(this);
        return true;
    }
}